src/main/java/core/basesyntax/dao/FruitDao.java

```diff
@@ -0,0 +1,76 @@
+package core.basesyntax.dao;
+
+import java.util.Objects;
+
+public class FruitDao {
+    private String name;
+    private int quantity;
+
+    public FruitDao(String name, int quantity) {
+        this.name = name;
+        this.quantity = quantity;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public void addQuantity(int amount) {
+        if (amount < 0) {
+            throw new IllegalArgumentException("Cannot add negative quantity.");
+        }
+        this.quantity += amount;
+    }
+
+    public void subtractQuantity(int amount) {
+        if (amount < 0) {
+            throw new IllegalArgumentException("Cannot subtract negative quantity.");
+        }
+        if (this.quantity < amount) {
+            throw new IllegalStateException("Not enough quantity to subtract.");
```

Comment:
Ensure that exceptions are handled properly and provide meaningful messages to help with debugging. Consider logging the exception or providing more context in the error message.